dolphinscheduler-python/pydolphinscheduler/src/pydolphinscheduler/core/resource.py

```diff
@@ -19,9 +19,21 @@
 
 from typing import Optional
 
+from pydolphinscheduler.java_gateway import launch_gateway
 from pydolphinscheduler.models import Base
 
 
+def query_resource_id(user_name, full_name):
+    """Get resource id from java gateway."""
+    return query_resource(user_name, full_name).getId()
+
+
+def query_resource(user_name, full_name):
+    """Get resource info from java gateway, contains resource id, name."""
+    gateway = launch_gateway()
+    return gateway.entry_point.queryResourcesFileInfo(user_name, full_name)
```

Comment:
I am not sure about that, but can we make it more OOP, and can we migrate function `query_resource_id ` and `query_resource ` into class `Resource`? like

```py
resource = Resource(user_name='user', path='path/to/resource')
resource.get_from_database()
resource_id = resource.resource_id
```

I task a look to current database from java code, and find out we have both `t_ds_resources` to store resource and `t_ds_relation_resources_user` to store relationship between user and resource, So in java code we using two object to store POJO and relation between, should we combine then into single model in Python API?

I want to hear your idea about this point @hiSandog

Reply:
And maybe you're doing some thing just like that @lyleshaw

Reply:
> And maybe you're doing some thing just like that @lyleshaw, and what do you think?

Sorry, I haven't dealt with the resource part of the code yet, but I think it's a good idea to combine them.